logger/common.go

```diff
@@ -334,6 +371,18 @@ func (l *Logger) Read(
 	}
 }
 
+// generateRandomID is based on Docker
+// GenerateRandomID: https://github.com/moby/moby/blob/bca8d9f2ce0d63e1490692917cde6273bc288bad/pkg/stringid/stringid.go#L40
+// with the simplification that we don't need to worry about guaranteeing the string isn't all 0 - 9 
+func generateRandomID() string {
+	b := make([]byte, 32)
+	if _, err := rand.Read(b); err != nil {
+		panic(err) // This shouldn't happen
+	}
```

Comment:
Why do we not expect that to happen?

Reply:
I copied this from docker, I guess it is bad code... I can change it...

Reply:
> On Linux, FreeBSD, Dragonfly and Solaris, Reader uses getrandom(2) if available, /dev/urandom otherwise. On OpenBSD and macOS, Reader uses getentropy(2). On other Unix-like systems, Reader reads from /dev/urandom. On Windows systems, Reader uses the RtlGenRandom API. On Wasm, Reader uses the Web Crypto API.

I think it only errors if the underlying syscall fails. But yea.. probably should handle it just in case..